waspc/test/Generator/PackageJsonGeneratorTest.hs

```diff
@@ -2,35 +2,70 @@ module Generator.PackageJsonGeneratorTest where
 
 import Test.Tasty.Hspec
 import qualified Wasp.AppSpec.App.Dependency as D
-import Wasp.Generator.PackageJsonGenerator (resolveNpmDeps)
+import Wasp.Generator.PackageJsonGenerator
+  ( DependencyConflictError (DependencyConflictError),
+    resolveDependencies,
+    resolveNpmDeps,
+  )
 
 spec_resolveNpmDeps :: Spec
 spec_resolveNpmDeps = do
   let waspDeps =
-        [ ("axios", "^0.20.0"),
-          ("lodash", "^4.17.15")
+        D.fromList
+          [ ("a", "1"),
+            ("b", "2")
+          ]
+
+  it "a conflicting version number is detected" $ do
+    let userDeps =
+          D.fromList
+            [ ("a", "1"),
+              ("b", "3")
+            ]
+    resolveDependencies waspDeps userDeps
+      `shouldBe` Left
+        [ DependencyConflictError
+            (D.create ("b", "2"))
+            (D.create ("b", "3"))
         ]
+  it "same versions so no conflict" $ do
```

Comment:
you also test here the following property: that user deps are not repeated if they are already in wasp deps. So it might be worth mentioning that in the test description.